adcc/OneParticleOperator.py

```diff
@@ -52,41 +57,183 @@ def __init__(self, spaces, is_symmetric=True):
             assert isinstance(spaces.reference_state, libadcc.ReferenceState)
             self.reference_state = spaces.reference_state
 
-    @classmethod
-    def from_cpp(cls, cpp_operator):
-        assert cpp_operator.cartesian_transform == "1"
-        ret = cls(cpp_operator.mospaces, cpp_operator.is_symmetric)
-        for b in cpp_operator.blocks_nonzero:
-            ret.set_block(b, cpp_operator.block(b))
+        occs = sorted(self.mospaces.subspaces_occupied, reverse=True)
+        virts = sorted(self.mospaces.subspaces_virtual, reverse=True)
+        self.orbital_subspaces = occs + virts
+        if self.is_symmetric:
+            combs = list(
+                combinations_with_replacement(self.orbital_subspaces, r=2)
+            )
+        else:
+            combs = list(product(self.orbital_subspaces, repeat=2))
+        self.blocks = ["".join(com) for com in combs]
+        self._tensors = {}
+
+    @property
+    def shape(self):
+        """
+        Returns the shape tuple of the OneParticleOperator
+        """
+        size = sum(self.mospaces.n_orbs(ss) for ss in self.orbital_subspaces)
+        return (size, size)
+
+    @property
+    def size(self):
+        """
+        Returns the number of elements of the OneParticleOperator
+        """
+        return np.prod(self.shape)
+
+    @property
+    def blocks_nonzero(self):
+        """
+        Returns a list of the non-zero block labels
+        """
+        return [b for b in self.blocks if b in self._tensors]
+
+    def is_zero_block(self, block):
+        """
+        Checks if block is explicitly marked as zero block.
+        Returns False if the block does not exist.
+        """
+        if block not in self.blocks:
+            return False
+        return block not in self.blocks_nonzero
+
+    def block(self, block):
+        """
+        Returns tensor of the given block
```

Comment:
Add: Does not create a block in case it is marked as a zero block. Use `__getitem__` for that purpose.